### ISSUE TITLE
Subclassing not working as expected

### ISSUE THREAD

**Issue by rserbitar** [pydata/pandas#11559]
Running the code:

```python
import pandas

class SubclassedSeries(pandas.Series):
    @property
    def _constructor(self):
        return SubclassedSeries

    @property
    def _constructor_expanddim(self):
        return SubclassedDataFrame

    def c(self):
        return 1

class SubclassedDataFrame(pandas.DataFrame):
    @property
    def _constructor(self):
        return SubclassedDataFrame

    @property
    def _constructor_sliced(self):
        return SubclassedSeries


b = SubclassedDataFrame([[1,2,3,4],[2,3,4,5]])
print(b[0].c())
```

works as expected.

However running:

```python
print(b.ix[1].c())
```
throws an exception as:

```python
b.ix[1]
```
returns a Series instance instead of an SublcassedSeries instance as expected.

**Comment by jreback** [comment]
ok, subclassing is not recommended, see docs [here](http://pandas.pydata.org/pandas-docs/stable/internals.html#subclassing-pandas-data-structures), but would take a pull-request to fully test out the indexers (as this is not currently done).

**Comment by sinhrks** [comment]
This is already fixed on master (maybe by #12787). Needs tests though.

**Comment by kielpins-hp** [comment]
The lack of subclassing makes it difficult and/or fragile to extend the package in any way. It's really unfortunate.
